lib/exercism/excerpt.rb

```diff
@@ -0,0 +1,26 @@
+class Excerpt
+  def initialize(text)
+    @html_text = text
+  end
+
+  def limit(length)
+    text[0..(length - 1)] + ellipsis(length)
+  end
+
+  private
+
+  def text
+    @_raw_text ||= begin
+      document = Nokogiri::HTML(@html_text)
+      remove_whitespace(document.text)
+    end
+  end
+
+  def remove_whitespace(string)
+    string.split("\n").map(&:strip).join(' ').strip
+  end
+
+  def ellipsis(length)
+    text.length < length ? '' : '...'
```

Comment:
We should be using the `&hellip;` HTML entity rather than three periods here.

This might cause a problem, though, because we now combine an HTML unsafe string (their abbreviated content) and the `&hellip;`, which _is_ safe. This might not render properly in the view.